doc/api/esm.md

```diff
@@ -396,59 +404,72 @@ Named exports detection covers many common export patterns, reexport patterns
 and build tool and transpiler outputs. See [cjs-module-lexer][] for the exact
 semantics implemented.
 
-## Builtin modules
+### Differences between ES modules and CommonJS
 
-[Core modules][] provide named exports of their public API. A
-default export is also provided which is the value of the CommonJS exports.
-The default export can be used for, among other things, modifying the named
-exports. Named exports of builtin modules are updated only by calling
-[`module.syncBuiltinESMExports()`][].
+#### No `require`, `exports` or `module.exports`
 
-```js
-import EventEmitter from 'events';
-const e = new EventEmitter();
-```
+In most cases, the ES module `import` can be used to load CommonJS modules.
+
+If needed, a `require` function can be constructed within an ES module using
+[`module.createRequire()`][].
 
+#### No `__filename` or `__dirname`
+
+These CommonJS variables are not available in ES modules.
+
+`__filename` and `__dirname` use cases can be replicated via
+[`import.meta.url`][].
+
+#### No JSON Module Loading
+
+JSON imports are still experimental and only supported via the
+`--experimental-json-modules` flag.
+
+Local JSON files can be loaded relative to `import.meta.url` with `fs` directly:
+
+<!-- eslint-skip -->
 ```js
-import { readFile } from 'fs';
-readFile('./foo.txt', (err, source) => {
-  if (err) {
-    console.error(err);
-  } else {
-    console.log(source);
-  }
-});
+import { readFile } from 'fs/promises';
+const json = JSON.parse(await readFile(new URL('./dat.json', import.meta.url)));
 ```
 
-```js
-import fs, { readFileSync } from 'fs';
-import { syncBuiltinESMExports } from 'module';
+Alterantively `module.createRequire()` can be used.
 
-fs.readFileSync = () => Buffer.from('Hello, ESM');
-syncBuiltinESMExports();
+#### No Native Module Loading
 
-fs.readFileSync === readFileSync;
-```
+Native modules are not currently supported with ES module imports.
 
-## CommonJS, JSON, and native modules
+The can instead be loaded with `module.createRequire()` or `process.dlopen`.
```

Comment:
I think it would be helpful to have them as links.
```suggestion
The can instead be loaded with [`module.createRequire()`][] or [`process.dlopen`][].
```